### PR TITLE
Experimental flags in flutter

### PR DESCRIPTION
**Pull request by liamappelbe**
Adds a command line flag to the flutter tool for experimental flags. They are passed through extra-front-end-options to simplify some of the plumbing.

https://github.com/flutter/flutter/issues/26240

### CODE REVIEW

**Review by a-siva — APPROVED** [review]

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -389,6 +390,22 @@ abstract class FlutterCommand extends Command<void> {
...
     }
 
+    String extraFrontEndOptions =
+        argParser.options.containsKey(FlutterOptions.kExtraFrontEndOptions)
+            ? argResults[FlutterOptions.kExtraFrontEndOptions]
+            : null;
+    if (argParser.options.containsKey(FlutterOptions.kEnableExperiment) &&
+        argResults[FlutterOptions.kEnableExperiment] != null) {
+      for (String expFlag in argResults[FlutterOptions.kEnableExperiment]) {
+        final String flag = '--enable-experiment=' + expFlag;
```
[the remark this answers is not in the document]
Comment (liamappelbe) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: That was my initial approach, but extra-front-end-options is also delimited by ','. So "--extra-front-end-options=--enable-experiment=abc,def" would be passed to the front end as 2 separate flags: "--enable-experiment=abc" and "def"

[the remark this answers is not in the document]
Comment (a-siva) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: Thanks for the clarification.

**Comment by alexmarkov** [comment]
1. Do we need to pass these new --enable-experiment options to gen_snapshot and/or VM?

2. Also, please double check that --extra-front-end-options works in case of incremental compilation (in debug mode, when doing hot reload).

**Comment by liamappelbe** [comment]
> 1. Do we need to pass these new --enable-experiment options to gen_snapshot and/or VM?

VM part is [done](https://dart-review.googlesource.com/c/sdk/+/89042), and gen_snapshot will be in a follow up CL.

> 2. Also, please double check that --extra-front-end-options works in case of incremental compilation (in debug mode, when doing hot reload).

Will do

**Comment by alexmarkov** [comment]
> > 1. Do we need to pass these new --enable-experiment options to gen_snapshot and/or VM?
> 
> VM part is [done](https://dart-review.googlesource.com/c/sdk/+/89042), and gen_snapshot will be in a follow up CL.

If VM needs to look at those flags, Flutter embedder should also pass these flags to Dart VM.

**Comment by liamappelbe** [comment]
> > > 1. Do we need to pass these new --enable-experiment options to gen_snapshot and/or VM?
> > 
> > 
> > VM part is [done](https://dart-review.googlesource.com/c/sdk/+/89042), and gen_snapshot will be in a follow up CL.
> 
> If VM needs to look at those flags, Flutter embedder should also pass these flags to Dart VM.

The plan is for the front end to save the flags in the dill file, and for the VM to only get the flags from there. It doesn't need them directly.

**Review by alexmarkov — APPROVED** [review]
lgtm

**Pull request merged**